### PR TITLE
18089436 integration type links

### PR DESCRIPTION
**Pull request by Adorism**
## Change description

This PR adds internal links to some documents that allows users to jump to a specific source or destination type from the table listing. It also standardizes the notes section using the notes component for previously hand-written docs. There are also updates to the explanation of tagging and list creation behavior for some destinations. 

### Development

- [x] Application changes have been tested appropriately

### Impact

- [ ] Code follows company security practices and guidelines
- [ ] Security impact of change has been considered
- [ ] Performance impact of change has been considered
- [ ] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: pages/docs/integrations/grouparoo-braze.mdx [hunk 1]
```diff
@@ -24,7 +24,9 @@ Braze is a customer engagement platform. Braze allows you to leverage contextual
   </tr>
   <tr>
     <td>
-      <strong>Braze Export Users</strong>
+      <a href="#braze-export-users">
+        <strong>Braze Export Users</strong>
```
Comment (teallarson) [inline pages/docs/integrations/grouparoo-braze.mdx]: I see some docs use bold text for this column and some use regular weight.  Does the bold text mean something?

File: pages/docs/integrations/grouparoo-sailthru.mdx [hunk 2]
```diff
@@ -20,7 +20,9 @@ Sailthru is a cloud-based marketing automation service. You can use Sailthru to
       <td>Description</td>
     </tr>
     <tr>
-      <td>Sailthru Export Users</td>
+      <a href="#sailthru-export">
```
Comment (teallarson) [inline pages/docs/integrations/grouparoo-sailthru.mdx]: broken link

**Review by teallarson — COMMENTED** [review]
Looking good!  One broken link and a styling question.

File: pages/docs/integrations/grouparoo-braze.mdx (hunk 1 above)
Comment (Adorism) [reply pages/docs/integrations/grouparoo-braze.mdx]: Good Question. That's something I have been trying to keep standard. In the overview tables it is less meaningful and maybe this is where I have been less consistent. In other tables I try to bold anything that is a field name.

File: pages/docs/integrations/grouparoo-sailthru.mdx (hunk 2 above)
Comment (Adorism) [reply pages/docs/integrations/grouparoo-sailthru.mdx]: good catch

File: pages/docs/integrations/grouparoo-braze.mdx (hunk 1 above)
Comment (Adorism) [reply pages/docs/integrations/grouparoo-braze.mdx]: I think that I will probably stick to having the destination and source type names at regular weight. Will change this.

**Review by teallarson — APPROVED** [review]
Yeah docs!

**Pull request merged**